### ISSUE TITLE
resource skipped due to guards are not counted

### ISSUE THREAD

**Issue by alexpop** [chef/chef#3581]
I have the following code in a recipe:

```ruby
bash 'test1' do
  code 'touch /tmp/test.txt'
  creates "/tmp/test.txt"
end

bash 'test2' do
  code 'touch /tmp/test.txt'
  not_if { File.exists?('/tmp/test.txt') }
end
```

And it produces this output using chef-client 12.3.0:

```
Converging 2 resources
Recipe: test::sandbox
  * bash[test1] action run (up to date)
  * bash[test2] action run (skipped due to not_if)

Running handlers:
Running handlers complete
Chef Client finished, 0/1 resources updated in 1.571081499 seconds
```

Since two resources got added into the resource collection, I'm expecting the last line to be:
```
Chef Client finished, 0/2 resources updated
```

Both resources in my example got skipped due to idempotency control attributes, but the one using not_if was not counted.

**Comment by juliandunn** [comment]
Per the linked PR, this should be fixed.

**Comment by alexpop** [comment]
This was not included in chef-client 12.4.1, but I see the CHANGELOG.md referencing this fix under 12.5.0

Thank you!
